### PR TITLE
fix(testing-library): render options type

### LINKED ISSUE #1462: [Bug]: [testing-library] Incompatible type of `RenderHookOptions.wrapper.children`
### System Info

```
  System:
    OS: macOS 15.3.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 84.56 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.15.0 - ~/n/bin/node
    npm: 10.9.2 - ~/n/bin/npm
    pnpm: 10.6.5 - ~/n/bin/pnpm
    bun: 1.0.26 - ~/.bun/bin/bun
    Watchman: 2025.04.28.00 - /opt/homebrew/bin/watchman
  npmPackages:
    @lynx-js/qrcode-rsbuild-plugin: ^0.4.0 => 0.4.0 
    @lynx-js/react: ^0.112.1 => 0.112.1 
    @lynx-js/react-rsbuild-plugin: ^0.10.10 => 0.10.10 
    @lynx-js/rspeedy: ^0.10.5 => 0.10.5 
    @lynx-js/types: 3.3.0 => 3.3.0 
```

### Details

Hello. I took an example from [the docs](https://lynxjs.org/api/reactlynx-testing-library/function.renderhook) of how to use the `renderHook` and it seems like the TS complains regarding the type of `RenderHookOptions.wrapper.children`:

<details>
<summary>TS error</summary>

```
src/__tests__/index.test.tsx:24:49 - error TS2322: Type 'LynxElement & ComponentChildren' is not assignable to type 'ReactNode'.
  Type 'LynxElement & bigint' is not assignable to type 'ReactNode'.
    Type 'LynxElement & BigInt' is missing the following properties from type 'ReactPortal': type, props, key

24       return <Context.Provider value='provided'>{children}</Context.Provider>;
                                                   ~~~~~~~~~~

  node_modules/@types/react/index.d.ts:618:9
    618         children?: ReactNode | undefined;
                ~~~~~~~~
    The expected type comes from property 'children' which is declared here on type 'IntrinsicAttributes & ProviderProps<string>'


Found 1 error in src/__tests__/index.test.tsx:24
```
</details>

### Reproduce link

https://github.com/plesiecki/rspeedy-renderhook-wrapper-type

### Reproduce Steps

1. `git clone https://github.com/plesiecki/rspeedy-renderhook-wrapper-type && cd rspeedy-renderhook-wrapper-type`
2. `npm i`
3. `npx tsc`
4. Notice the error

### PR DESCRIPTION
**Pull request by upupming**
Fixes #1462

## Summary by CodeRabbit

- Bug Fixes
  - Fixed type errors affecting the testing-library render and renderHook wrapper option and improved React type compatibility.

- Tests
  - Added type-level tests validating render and renderHook behavior, including wrapper and context scenarios.

- Chores
  - Added a changeset for a patch release, added a type-check-only test script in examples, and updated example JSX runtime for type checking.

## Checklist

- [x] Tests updated (or not required).
- [ ] Documentation updated (or **not required**).
- [x] Changeset added, and when a BREAKING CHANGE occurs, it needs to be clearly marked (or not required).

### CODE REVIEW

**Review by colinaaa — APPROVED** [review]
LGTM. But I think the coderabbitai makes sense on choosing between `ReactNode` and `ReactElement`. We may want more tests to show the differences.

File: packages/react/testing-library/types/entry.d.ts [hunk 1]
```diff
@@ -69,7 +68,7 @@ export interface RenderOptions<Q extends Queries = typeof queries> {
    * export { customRender as render }
    * ```
    */
-  wrapper?: ComponentChild;
+  wrapper?: React.JSXElementConstructor<{ children: React.ReactNode }> | undefined;
```
[the remark this answers is not in the document]
Comment (upupming) [reply packages/react/testing-library/types/entry.d.ts:71]: > Requiring children as a mandatory prop is too restrictive

Wrapper must have a `children` because it is used to wrap the component to be rendered, it will be used like:

```js
h(WrapperComponent, null, innerElement)
```

File: packages/react/testing-library/types/entry.d.ts [hunk 2]
```diff
@@ -95,7 +94,7 @@
  */
 export type RenderResult<Q extends Queries = typeof queries> = {
   container: LynxElement;
-  rerender: (ui: ComponentChild) => void;
+  rerender: (ui: React.ReactNode) => void;
```
[the remark this answers is not in the document]
Comment (upupming) [reply packages/react/testing-library/types/entry.d.ts:97]: "accepting strings/null/etc" is what the React Testing Library does. See https://github.com/testing-library/react-testing-library/blob/1c931a6c03091d725eccee7767d9ec696d5d33c6/types/index.d.ts#L180

Let me add some type testing on pure "strings/null/etc"

File: packages/react/testing-library/types/entry.d.ts [hunk 3]
```diff
@@ -132,7 +131,7 @@
  * @public
  */
 export function render<Q extends Queries>(
-  ui: ComponentChild,
+  ui: React.ReactNode,
   options?: RenderOptions<Q>,
 ): RenderResult<Q>;
```
[the remark this answers is not in the document]
Comment (upupming) [reply packages/react/testing-library/types/entry.d.ts:136]: same as above

File: packages/react/testing-library/types/entry.d.ts [hunk 4]
```diff
@@ -213,7 +212,7 @@
    * export { customRender as renderHook }
    * ```
    */
-  wrapper?: ComponentType<{ children: LynxElement }>;
+  wrapper?: React.JSXElementConstructor<{ children: React.ReactNode }> | undefined;
```
[the remark this answers is not in the document]
Comment (upupming) [reply packages/react/testing-library/types/entry.d.ts:215]: same as above

**Comment by upupming** [comment]
> LGTM. But I think the coderabbitai makes sense on choosing between `ReactNode` and `ReactElement`. We may want more tests to show the differences.

Tests are added, PTAL~

**Review by colinaaa — APPROVED** [review]
👍
